cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -1352,22 +1356,22 @@ nextEpochEligibleLeadershipSlots sbe sGen serCurrEpochState _ptclState
       then return ()
       else Left $ LeaderErrStakeDistribUnstable tip stableStakeDistribSlot stabilityWindowSlots currentEpochLastSlot
 
-  -- Let's do a nonce check. The candidate nonce and the evolving nonce should not be equal.
   chainDepState <- first LeaderErrDecodeProtocolStateFailure
-                     $ error "nextEpochEligibleLeadershipSlots: Currently broken" -- decodeProtocolState ptclState
+                     $ decodeProtocolState ptclState
 
   -- We need the candidate nonce, the previous epoch's last block header hash
   -- and the extra entropy from the protocol parameters. We then need to combine them
   -- with the (⭒) operator.
-  let TPraos.PrtclState _ evolvingNonce candidateNonce = TPraos.csProtocol chainDepState
+  let Consensus.PraosNonces { Consensus.candidateNonce, Consensus.evolvingNonce } =
+        Consensus.getPraosNonces (Proxy @(ConsensusProtocol era)) chainDepState
 
   when (evolvingNonce == candidateNonce)
    $ Left LeaderErrCandidateNonceStillEvolving
 
   -- Get the previous epoch's last block header hash nonce
-  let Tick.TicknState _ prevEpochLastBlkHeaderHashNonce = TPraos.csTickn chainDepState
+  let previousLabNonce = Consensus.previousLabNonce (Consensus.getPraosNonces (Proxy @(ConsensusProtocol era)) chainDepState)
```

Comment:
Please check.

Reply:
We've had

```
data TicknState = TicknState
  { ticknStateEpochNonce :: !Nonce,
    ticknStatePrevHashNonce :: !Nonce
  }
```

and now we also have

```
instance SL.PraosCrypto c => PraosProtocolSupportsNode (TPraos c) where
  getPraosNonces _prx cdst =
      PraosNonces {
          candidateNonce
        , epochNonce       = ticknStateEpochNonce
        , evolvingNonce
        , labNonce         = csLabNonce
        , previousLabNonce = ticknStatePrevHashNonce
        }
    where
      TPraosState { tpraosStateChainDepState } = cdst
      SL.ChainDepState {
          SL.csLabNonce
        , SL.csProtocol
        , SL.csTickn
        } = tpraosStateChainDepState
      SL.PrtclState
        _opcertCounters
        evolvingNonce
        candidateNonce
          = csProtocol
      SL.TicknState {
          ticknStateEpochNonce
        , ticknStatePrevHashNonce
        } = csTickn
```

so, yes, I think `Consensus.previousLabNonce` is indeed the new name of the second argument of `TicknState`.

Reply:
Thanks!